portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,29 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = randomFact;
 }
+
+function getFromServlet(){
+    console.log("Fetching from sevlet");
+
+    const fetchPromise = fetch('/data');
+
+    fetchPromise.then(handle);
+}
+
+function handle(response){
```

Comment:
you can use arrow functions to put this all inline.  It's a lot easier to read that way, and unless you have a reason to think you'll reuse these in the future, there's not much benefit to making separate functions

Reply:
Did you mean for all three functions? I'm currently still a bit confused by the arrow functions but I will continue studying them and then update.

Reply:
I meant handle and addToSite.  getFromServlet is used in your HTML, so it has to be named.  This is a good time to try it out, let me know if you have any questions

Reply:
Okay, I think I understand how to use arrow functions now. But, I've edited the addToSite function so that it can also handle an error. Would I be able to do that with arrow functions?

Reply:
yea, you can put any code you want inside an an arrow function.  Just note that you'll need curly braces if you're doing anything multi-line

Reply:
I've put all the code in to the arrow function from handle and addToSite but nothing appears and I get no error.